### PR TITLE
Added support for msvc builds in cpp-qt5-client generator

### PR DESCRIPTION
**Pull request by onze**
Moved GCC-specific compile flags to non msvc builds, and added equivalent flags for msvc.

Copying the c++ technical committee @ravinikam @stkrwork @etherealjoy @martindelille @muttleyxd

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/cpp-qt5-client/CMakeLists.txt.mustache [hunk 1]
```diff
@@ -5,7 +5,11 @@ set(CMAKE_VERBOSE_MAKEFILE ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_AUTOMOC ON)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -Wall -Wno-unused-variable")
+if (MSVC)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4")
+else (MSVC)
```
Comment (muttleyxd) [inline modules/openapi-generator/src/main/resources/cpp-qt5-client/CMakeLists.txt.mustache]: I'd go with
```
else ()
...
endif ()
```

to make it look more like other programming languges.

Comment (onze) [reply modules/openapi-generator/src/main/resources/cpp-qt5-client/CMakeLists.txt.mustache]: Got it, thanks

**Comment by wing328** [comment]
Drone.io CI failure not related to this PR.

**Comment by wing328** [comment]
@onze thanks for the PR, which has been included in the 4.3.0 release: https://twitter.com/oas_generator/status/1243455743937789952

**Pull request merged**